### DISCUSSION TITLE
freeplane-io Python library for external scripting

### DISCUSSION THREAD

**Discussion by nnako** [freeplane/freeplane#697] [Show and tell]
Hi folks,

I proudly present the `freeplane-python-io` project / package which makes possible the "external" access of Freeplane mindmaps using the Python language. Different from the approach using Groovy-scripting *within* the Freeplane editor, the concept here is to read and manipulate the MM files directly, without even starting the Freeplane editor.

In addition to designing mindmaps using the graphical user interface (GUI) which Freeplane provides marvelously, it is now possible to `automate` most of the searching, browsing, reading, changing, creating of nodes by means of scripting in the Python language in a kind of "head-less" fashion, directly on the command line or integrated within your own applications.

I started this project in 2016 when I realized that I rather wanted to **let my scripts operate** on my big Freeplane mindmaps - why should I sit in front of the GUI myself? ;-)

The usage is easy:

- install the package into your local Python environment by typing `pip install freeplane-io`
- open your Python console with your environment activated by typing `python`
- import the package by typing `import freeplane` within the Python REPL
- then use the package to load / create / manage / save mindmaps
- ...

----

package home: https://pypi.org/project/freeplane-io/
project home: https://github.com/nnako/freeplane-python-io

----

**Unfortunately, I didn't manage to provide a good documentation, yet. So, everybody has to look up the features directly within the source code (currently, just one large Python file within the src folder). Sorry for that. I will provide proper documentation very soon.**

Any contribution is very welcome!

**Comment by macmarrum** [comment]
Cool :+1:
Thanks

**Reply by nnako** [reply]
you are very welcome. further documentation will follow, as I promised you in our conversation last Saturday.
